openeo/rest/connection.py

```diff
@@ -1385,6 +1385,56 @@ def load_stac(
             _log.warning(f"Failed to extract cube metadata from STAC URL {url}", exc_info=True)
         return cube
 
+    def load_stac_from_job(
+        self,
+        job: Union[BatchJob, str],
+        spatial_extent: Optional[Dict[str, float]] = None,
+        temporal_extent: Union[Sequence[InputDate], Parameter, str, None] = None,
+        bands: Optional[List[str]] = None,
+        properties: Optional[Dict[str, Union[str, PGNode, Callable]]] = None,
+    ) -> DataCube:
+        """
+        Wrapper for :py:meth:`load_stac` that loads the result of a previous job using the STAC collection of its results.
+
+        :param job: The batch job object that has successfully finished, or a job id. When a job id is provided,
+            a job object is created using self as connection.
+        :param spatial_extent: limit data to specified bounding box or polygons
+        :param temporal_extent: limit data to specified temporal interval.
+        :param bands: only add the specified bands
+        :return: a :py:class:`DataCube`
+        """
+        if isinstance(job, str):
+            job = BatchJob(job_id=job, connection=self)
+        elif not isinstance(job, BatchJob):
+            raise ValueError("job must be a BatchJob or job id")
+
+        try:
+            job_results = job.get_results()
+
+            canonical_links = [
+                link["href"] for link in job_results.get_metadata()["links"] if link["rel"] == "canonical"
+            ]
+            if len(canonical_links) == 0:
+                _log.warning("No canonical link found in job results metadata. Using job results URL instead.")
+                stac_link = job.get_results_metadata_url(full=True)
+            else:
+                if len(canonical_links) > 1:
+                    _log.warning(
+                        f"Multiple canonical links found in job results metadata: {canonical_links}. Picking first one."
+                    )
+                stac_link = canonical_links[0]
+        except OpenEoApiError as e:
```

Comment:
I think you should also catch exceptions from failing `link["href"]`  or `link["rel"]` attempts above

Reply:
When these fail, `canonical_links` will just me empty and the unsigned link will be used as well. I've added an extra test to cover this behaviour

Reply:
No I mean that the link dictionaries could be missing "rel" or "href" fields. Or maybe the link list items might not even dictionaries. Or the whole "links" field might be missing

It would actually be safer to write it like this:
```python
canonical_links = [
    link["href"] 
    for link in job_results.get_metadata().get("links", []) 
    if link.get("rel") == "canonical" and "href" in link
]
```